app/queries/drug_stocks_query.rb

```diff
@@ -0,0 +1,119 @@
+class DrugStocksQuery
+  def initialize(facilities:, for_end_of_month:)
+    @facilities = facilities
+    @for_end_of_month = for_end_of_month
+    # assuming that all facilities on the page have the same protocol
+    @protocol = @facilities.first.protocol
+  end
+
+  def protocol_drugs_by_category
+    @drug_categories ||= @protocol.protocol_drugs
+      .where(stock_tracked: true)
+      .order(:name, :dosage)
+      .sort_by {|protocol_drug| [protocol_drug.name, protocol_drug.dosage.to_i]}
+      .group_by(&:drug_category)
+      .sort_by {|(drug_category, _) | drug_category}
+      .to_h
+  end
+
+  def call
+    patient_counts = Patient.where(assigned_facility: @facilities.map(&:id)).group(:assigned_facility).count
```

Comment:
You shouldn't need to `map(&:id)` here.

Reply:
We tried that first, and I don't remember the exact error we ran into. @vkrmis?

Reply:
I think we were doing something slightly different then, `assigned_facility: @facilities` should work

Reply:
`PG::AmbiguousColumn: ERROR:  column reference "id" is ambiguous` is what I get when trying this.